azext_iot/tests/__init__.py

```diff
@@ -71,16 +86,57 @@ def command_execute_assert(self, command, asserts):
 
 
 class IoTLiveScenarioTest(CaptureOutputLiveScenarioTest):
-    def __init__(self, test_scenario, entity_name, entity_rg):
+    def __init__(self, test_scenario):
         assert test_scenario
-        assert entity_name
-        assert entity_rg
-
-        self.entity_name = entity_name
-        self.entity_rg = entity_rg
-
+        self.entity_rg = ENTITY_RG
+        self.entity_name = ENTITY_NAME
         super(IoTLiveScenarioTest, self).__init__(test_scenario)
 
+        if not settings.env.azext_iot_testhub:
+            hubs_list = self.cmd(
+                '''iot hub list -g "{}"'''.format(self.entity_rg)
+            ).get_output_in_json()
+
+            target_hub = None
+            for hub in hubs_list:
+                if hub["name"] == self.entity_name:
+                    target_hub = hub
+                    break
+
+            if not target_hub:
+                if settings.env.azext_iot_teststorageaccount:
+                    storage_account_connenction = self.cmd(
+                        "storage account show-connection-string --name {}".format(
+                            settings.env.azext_iot_teststorageaccount
+                        )
+                    ).get_output_in_json()
+
+                    self.cmd(
+                        "iot hub create --name {} --resource-group {} --fc {} --fcs {} --sku S1 ".format(
+                            self.entity_name, self.entity_rg,
+                            STORAGE_CONTAINER, storage_account_connenction["connectionString"]
+                        )
+                    )
+                else:
+                    self.cmd(
+                        "iot hub create --name {} --resource-group {} --sku S1 ".format(
+                            self.entity_name, self.entity_rg
+                        )
+                    )
+
+                profile = Profile(cli_ctx=DummyCli())
+                user_id = profile.get_current_account_user()
+                new_hub = self.cmd(
+                    "iot hub show -n {} -g {}".format(self.entity_name, self.entity_rg)
+                ).get_output_in_json()
+
+                # assign IoT Hub Data Contributor role to current user
+                self.cmd(
+                    '''role assignment create --assignee {} --role "{}" --scope "{}"'''.format(
```

Comment:
Role assignments commonly do not propagate immediately. We need to use any combination of wait timers and/or refresh account credentials. We commonly run into RBAC propagation issues on the DT tests side and are looking to improve the RBAC pattern there.

Reply:
Makes sense, although I didn't run into issues during my testing. But, it's better to be safe

Reply:
I am able to address this by refreshing accounts